src/System.Device.Gpio/System/Device/Spi/SpiConnectionSettings.cs

```diff
@@ -19,8 +19,8 @@ private SpiConnectionSettings()
         /// Initializes a new instance of the <see cref="SpiConnectionSettings"/> class.
         /// </summary>
         /// <param name="busId">The bus ID the device is connected to.</param>
-        /// <param name="chipSelectLine">The chip select line used on the bus.</param>
-        public SpiConnectionSettings(int busId, int chipSelectLine)
+        /// <param name="chipSelectLine">The chip select line used on the bus. Optional, -1 if not used</param>
+        public SpiConnectionSettings(int busId, int chipSelectLine = -1)
```

Comment:
if you make it optional here you should at least check if there is a way to support that on Linux (I think there might be), otherwise please leave as is and file a separate issue (might be a good idea anyway since that PR might need more work)

Reply:
From what I can find, passing in 0 seems to mean ignore ChipSelect on both Linux and Windows. Do you recommend we use 0 as the ignore value in the SoftwareSpi class as well, and change the default to that?

Reply:
no, 0 is a valid value for PhysicalNumbering numbering. As an implementation detail we can pass 0 to Linux struct when user passes -1, For 0 I'd pass through and I'm not opinionated if we should throw or just let it work so IMO just let it work unless we got some other opinions (@joperezr?).

Reply:
ok, I've added an implementation detail to both UnixSpiDevice and Win10SpiDevice to check for -1 and pass 0 to the OS.